pkg/command/login.go

```diff
@@ -41,397 +33,109 @@ var loginCmd = &cobra.Command{
-	loginCmd.Flags().MarkHidden("staging")     //nolint
+	utils.PanicOnErr(loginCmd.Flags().MarkHidden("api-token"))
+	utils.PanicOnErr(loginCmd.Flags().MarkHidden("staging"))
 	loginCmd.SetUsageFunc(cli.SubCmdUsageFunc)
 	loginCmd.MarkFlagsMutuallyExclusive("endpoint-ca-certificate", "insecure-skip-tls-verify")
 
-	// TODO: Update the plugin-runtime library with the new format and use the library method
-	msg := fmt.Sprintf("this was done in the %q release, it will be removed following the deprecation policy (6 months). Use the %q command instead.\n", "v0.90.0", "context")
-	loginCmd.Deprecated = msg
-
+	// TODO(prkalle): update the Note to point to login with TANZU_API_TOKEN and interactively
 	loginCmd.Example = `
-    # Login to TKG management cluster using endpoint
-    tanzu login --endpoint "https://login.example.com"  --name mgmt-cluster
-
-    #  Login to TKG management cluster by using the provided CA Bundle for TLS verification:
-    tanzu login --endpoint https://k8s.example.com[:port] --endpoint-ca-certificate /path/to/ca/ca-cert
-
-    # Login to TKG management cluster by explicit request to skip TLS verification, which is insecure:
-    tanzu login --endpoint https://k8s.example.com[:port] --insecure-skip-tls-verify
-
-    # Login to TKG management cluster by using kubeconfig path and context for the management cluster
-    tanzu login --kubeconfig path/to/kubeconfig --context path/to/context --name mgmt-cluster
-
-    # Login to TKG management cluster by using default kubeconfig path and context for the management cluster
-    tanzu login  --context path/to/context --name mgmt-cluster
-
-    # Login to an existing server
-    tanzu login --server mgmt-cluster
-
-    Note: To create Mission Control (TMC) contexts, an API Key is required. The API Key value can
-    be provided with the environment variable TANZU_API_TOKEN or can be entered as input while
-    creating the context.
-
-    [*] : Users have two options to login to TKG. They can choose the login endpoint option
-    by providing 'endpoint', or can choose to use the kubeconfig for the management cluster by
-    providing 'kubeconfig' and 'context'. If only '--context' is set and '--kubeconfig' is not,
-    the $KUBECONFIG env variable will be used and, if the $KUBECONFIG env is also not set, the
-    default kubeconfig file ($HOME/.kube/config) will be used.`
+    # Login to Tanzu
+    tanzu login
+
+    # Login to Tanzu using endpoint
+    tanzu login --endpoint "https://login.example.com"
+
+    #  Login to Tanzu by using the provided CA Bundle for TLS verification:
+    tanzu login --endpoint https://test.example.com[:port] --endpoint-ca-certificate /path/to/ca/ca-cert
+
+    # Login to Tanzu by explicit request to skip TLS verification (this is insecure)::
+    tanzu login --endpoint https://test.example.com[:port] --insecure-skip-tls-verify
+
+    Note:
+       To login to Tanzu an API Key is optional. If provided using the TANZU_API_TOKEN environment
```

Comment:
iirc the reason for hiding was because it is not good practice to provide the value of the token in as a command-line argument that might get captured in command history

Reply:
Yes, the reason is what @vuil mentioned.

Reply:
I wonder why have the flag at all then but I assume there must have been some request from some users. That’s fine. Thanks for the explanation.